internal/batches/executor/coordinator.go

```diff
@@ -199,48 +191,37 @@ func (c *Coordinator) cacheAndBuildSpec(ctx context.Context, taskResult taskResu
 	}
 
 	// Build the changeset specs.
-	specs, err = createChangesetSpecs(taskResult.task, taskResult.result, c.opts.Features)
+	specs, err := createChangesetSpecs(taskResult.task, taskResult.result, c.opts.Features)
 	if err != nil {
-		return specs, err
+		return nil, err
 	}
 
+	ui.TaskChangesetSpecsBuilt(taskResult.task, specs)
 	return specs, nil
 }
 
-type executionProgressPrinter func([]*TaskStatus)
+type TaskExecutionUI interface {
```

Comment:
This is the new interface. It's in `executor` because otherwise we'd have a circular reference between `ui` and `executor` since it mentions `*Task`. And before I remove `*Task` from this interface I want to find out what data we need. Then we can narrow it down.